### DISCUSSION TITLE
Duplicate an item

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#6501] [Archive]
Is there any way the user can duplicate an item?
In my case I have an item that doesn't differ much, if we could duplicate and change the differences it would be perfect and would save a lot of time

Select the respective item and a duplicate button :)

Thanks!..

**Comment by rijkvanzanten** [comment]
In the meantime, you can open the item, make the change, and then use "Save as Duplicate":

<img width="517" alt="CleanShot 2021-06-25 at 14 23 29@2x" src="https://user-images.githubusercontent.com/9141017/123469352-f1142b00-d5c0-11eb-9985-7980904ab1d4.png">

**Reply by hugopvl** [reply]
Thank you very much, it helps a lot

**Reply by LasseRosenow** [reply]
The big problem with this approach for me is, that if I have a O2M relation and I duplicate the `O` Item, all the `M` items get moved to the new copy. So the old Item loses all its children.

I think it would be better if in case of a O2M relation, all children get copied aswell

**Reply by rijkvanzanten** [reply]
@LasseRosenow That's configurable through Settings > Data Model > (collection) at the bottom under Field Duplication 👍🏻

**Reply by ghost** [reply]
@rijkvanzanten I don't unterstand how to use the Item Duplication Fields in Data models ... Where can I find a documentation ?  Thank you so much!

**Reply by rijkvanzanten** [reply]
@chtitbidule Think of it as [the "fields" parameter](https://docs.directus.io/reference/api/query/#fields). You control what data is used to create a new item with 🙂 Whatever you select in that interface will be loaded the moment you hit save and duplicate to make the duplicate record with

**Reply by rijkvanzanten** [reply]
(cc @alesvaupotic for docs)

**Reply by ghost** [reply]
OK thank you @rijkvanzanten !

**Reply by sevetseh28** [reply]
@rijkvanzanten is this duplication feature available through API? Im trying to find a workaround to support content versioning through API. Thanks

**Reply by rijkvanzanten** [reply]
> @rijkvanzanten is this duplication feature available through API? Im trying to find a workaround to support content versioning through API. Thanks

The app simply does a `GET` and then a `POST` with the data that was returned in the `GET`

**Reply by rokdd** [reply]
> @rijkvanzanten is this duplication feature available through API? Im trying to find a workaround to support content versioning through API. Thanks

I am also trying to solve versioning by something like this: https://github.com/directus/directus/discussions/14429
